File: 108-convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        if len(nums) == 0:
            return None
        elif len(nums) == 1:
            return TreeNode(val=nums[0])
        
        mid = len(nums) // 2

        left = self.sortedArrayToBST(nums[:mid])
        right = self.sortedArrayToBST(nums[mid+1:])

        root = TreeNode(val=nums[mid])
        root.left = left
        root.right = right

        return root

File: 108-convert-sorted-array-to-binary-search-tree/convert-sorted-array-to-binary-search-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public TreeNode SortedArrayToBST(int[] nums) {
        if (nums.Length == 0)
            return null;
        
        int mid = nums.Length / 2; // Rounds down
        int[] left = nums.Skip(0).Take(mid).ToArray();
        int[] right = nums.Skip(mid + 1).Take(nums.Length - mid - 1).ToArray();

        TreeNode root = new TreeNode(nums[mid]);
        root.left = SortedArrayToBST(left);
        root.right = SortedArrayToBST(right);

        return root;
    }
}